presto-kafka/src/test/java/io/prestosql/plugin/kafka/TestKafkaIntegrationSmokeTest.java

```diff
@@ -160,7 +162,120 @@ private KafkaTopicFieldDescription createOneFieldDescription(String name, Type t
         return new KafkaTopicFieldDescription(name, type, mapping, null, dataFormat, null, false);
     }
 
-    @Test(dataProvider = "testRoundTripAllFormatsDataProvider")
+    @Test(dataProvider = "jsonDateTimeFormatsDataProvider")
+    public void testJsonDateTimeFormats(JsonDateTimeTestCase testCase)
+    {
+        assertUpdate("INSERT into write_test." + testCase.getTableName() +
+                " (" + testCase.getFieldNames() + ")" +
+                " VALUES " + testCase.getRowValues(), testCase.getNumRows());
+        MaterializedResult actual = computeActual(getSession(), "SELECT " + testCase.getFieldNames() + " FROM write_test." + testCase.getTableName() + " LIMIT " + testCase.getNumRows());
+        MaterializedResult expected = computeActual(getSession(), "VALUES " + testCase.getRowValues());
+        assertEquals(actual.getMaterializedRows(), expected.getMaterializedRows());
+    }
+
+    @DataProvider
+    public final Object[][] jsonDateTimeFormatsDataProvider()
+    {
+        return jsonDateTimeFormatsData().stream()
+                .collect(toDataProvider());
+    }
+
+    private List<JsonDateTimeTestCase> jsonDateTimeFormatsData()
+    {
+        return ImmutableList.<JsonDateTimeTestCase>builder()
```

Comment:
I think those test would benefit if stracture it more granurally, where single testcase defines 
 * format
 * column
 * expected value

Sth like this:
```java
        testCaseBuilder.
            .addTestCase("custom_date_time_json", "f_time", "TIME '01:02:03.456'")
            .addTestCase("custom_date_time_json", "f_time", "TIME '03:02:03.456'")
            .addTestCase("custom_date_time_json", "f_time_with_time_zone", "TIME '05:02:03.456 UTC'")
            .addTestCase("iso8601_json", "f_time", "TIME '05:02:03.456'")
```

I think it is easier to read as you have all information about single testcase in single line.
We can also focus more on data types which are more problematic and have more corner cases.

Under the hood we can insert whole row with all columns but one we care about set to null.

WDYT?

Reply:
On the other hand maybe the unit tests we have in `TestJsonDateTime` are enough and these should be treated as just smoke coverage?
@aalbu ?

Reply:
I agree this format would be much easier to read and debug. Could probably just use one table as well.

Reply:
@losipiuk I think lighter coverage here and testing all edge cases in `TestJsonDateTime` is a good idea.  Perhaps what we would need to add here would be some round-trip tests with legacy semantics enabled.  We should talk about that.

Reply:
I think that it would be better to test each individual format function. The current tests in `TestJsonDateTime` helped me figure out how to write the format functions, but they just aren't great tests.

Reply:
+1 to lighter smoke testing

Reply:
> we should talk about that

@aalbu see https://github.com/prestosql/presto/pull/4477#discussion_r458737542